File: frontend/src/api/community.ts
import request from '@/utils/request'

export interface CommunityPost {
  id: number
  student_id: number
  category: string
  title: string
  content: string
  view_count: number
  like_count: number
  comment_count: number
  created_at: string | null
  author_name: string | null
  author_avatar: string | null
  liked: boolean
}

export interface CommunityComment {
  id: number
  post_id: number
  student_id: number
  content: string
  created_at: string | null
  author_name: string | null
  author_avatar: string | null
}

export const COMMUNITY_CATEGORIES = ['技术', '提问', '分享', '求助']

export function getPosts(params?: { category?: string; q?: string; page?: number; page_size?: number; sort?: string }) {
  return request.get<CommunityPost[]>('/community/posts', { params })
}

export function getPost(id: number) {
  return request.get<CommunityPost>(`/community/posts/${id}`)
}

export function createPost(data: { category: string; title: string; content: string }) {
  return request.post<CommunityPost>('/community/posts', data)
}

export function updatePost(id: number, data: Partial<CommunityPost>) {
  return request.put<CommunityPost>(`/community/posts/${id}`, data)
}

export function deletePost(id: number) {
  return request.delete(`/community/posts/${id}`)
}

export function toggleLike(postId: number) {
  return request.post<{ liked: boolean; count: number }>(`/community/posts/${postId}/like`)
}

export function getComments(postId: number) {
  return request.get<CommunityComment[]>(`/community/posts/${postId}/comments`)
}

export function createComment(postId: number, content: string) {
  return request.post<CommunityComment>(`/community/posts/${postId}/comments`, { content })
}

export function deleteComment(commentId: number) {
  return request.delete(`/community/comments/${commentId}`)
}

export function getHotPosts(limit = 5) {
  return request.get<CommunityPost[]>('/community/hot', { params: { limit } })
}

File: backend/app/schemas/community.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class CommunityPostCreate(BaseModel):
    category: str = "分享"
    title: str
    content: str


class CommunityPostUpdate(BaseModel):
    category: str | None = None
    title: str | None = None
    content: str | None = None


class CommunityPostOut(BaseModel):
    id: int
    student_id: int
    category: str
    title: str
    content: str
    view_count: int = 0
    like_count: int = 0
    comment_count: int = 0
    created_at: datetime | None = None
    author_name: str | None = None
    author_avatar: str | None = None
    liked: bool = False

    model_config = ConfigDict(from_attributes=True)


class CommunityCommentCreate(BaseModel):
    content: str


class CommunityCommentOut(BaseModel):
    id: int
    post_id: int
    student_id: int
    content: str
    created_at: datetime | None = None
    author_name: str | None = None
    author_avatar: str | None = None

    model_config = ConfigDict(from_attributes=True)


class LikeOut(BaseModel):
    liked: bool
    count: int